src/repositories/events.ts

```diff
@@ -9,34 +10,35 @@ export type CreateEvent<D> = {
   readonly data: D;
 };
 
-export class EventRepository {
-  private readonly db: DB;
+export const eventRepository = {
+  createEvent<D>(event: CreateEvent<D>): Effect.Effect<DB, unknown, void> {
+    return Effect.gen(function* (_) {
+      const db = yield* _(DB);
+      return Effect.tryCatchPromise(
```

Comment:
primo esempio di interfacciamento con librerie che usano Promise.
Qui le scelte sono due:
- mappare ad `Effect` ogni singola operazione (super sbatti, ad occhio)
- lavorare con le `Promise` nel contesto di pg-promise e poi wrappare il tutto in `Effect.tryCatchPromise` (quello che ho fatto qui)

Il risultato finale non è malvagissimo, ma c'è un po' un salto da un API ad un altra (Effect.gen vs async/await)